src/statistics/Plugin.cc

```diff
@@ -0,0 +1,54 @@
+
```

Comment:
Mostly a high-level comment. I think this is very useful functionality. I have a very similar external plugin stashed away somewhere.

Given this is part of Zeek base, IMO, we can also track the queued times directly in the individual [EventHandler](https://github.com/zeek/zeek/blob/6dec7d6027eb52ad55699b5064238a5eefc9b317/src/EventHandler.h#L18-L20) instances rather than an external data structure.

The bif implementation would then enumerate all event_handler's and return the counts for each (potentially only the `!=0` ones).

I'd also suggest to avoid the "reset" functionality and move to a only-ever-incrementing counter model at least within the base and what the bif exposes. If a consumers are interested in relatives/deltas, they can add logic to compute them, but increasing counters are easier to work with.

Reply:
Can you think of any reason we can't just use the telemetry framework for this?

Reply:
> Can you think of any reason we can't just use the telemetry framework for this?

He, that's the right question I guess.. Mostly depends on which level we want to use it at: If telemetry framework in script-land, we'd still need to have a bif to get to the actual counters (or implement `new_event` and count the events, but that's probably not going to fly).

Or implement previously mentioned per-EventHandler counter with the C++ telemetry API. That should just work and will make them then available through script-land telemetry.

The two downsides I can think of with the latter:

* I believe C++ API generated metrics are currently unconditionally exposed via Prometheus and I'm unaware of knobs here. Not all setups may want to expose such a large number of metrics if they aren't going to look closely at them.

*   Using either, C++ API or script-land telemetry will make (with what we have today) Broker/CAF a hard dependency.

Reply:
>  Both approaches will make Broker/CAF a hard-dependency.

Well - we must have this today already because the SessionManager tracks connections using the C++ telemetry API (those stats are available through bifs, too, so it's maybe softer). I was mainly thinking about #2476 when stating this.

Reply:
Part of #2476 would be figuring out how to do a Broker-free telemetry manager. I just feel like if we have the framework there to track telemetry-type data, adding something new that tracks it in a different way seems sorta backwards. I guess it depends on if Microsoft would be able to use the Broker version of the telemetry data for what they're doing here.

Reply:
Alright I finally had a chance to look at the code around the comments here. I think I agree that it would be better for the individual event handlers to handle this themselves, but with one question. Would leaving it as a plugin make us able to disable the functionality (and thus the cost of keeping track) easier? If no, then it's definitely cleaner to let the event handlers take care of it as it also removes all of the plugin code and the extra hook.

Reply:
Opinionated - we can probably consider incrementing the metric (or just a C++ counter) per `QueueEvent()` on the handler cheap enough and circumvent the disable functionality question. I can't imagine a `++handler->times_queued` would show up in a profile. If the telemetry/metrics add too much overhead due to being fancier/thread-safe/... we'll probably need to look for improvements there.

Reply:
Makes sense to me. Let's go ahead and make this change as part of this PR.